### ISSUE TITLE
Quoted TypeAliases doesn't support multi line strings

### ISSUE THREAD

**Issue by Gobot1234** [microsoft/pyright#2275]
**Describe the bug**
pyright doesn't support any form of multi-line strings. Currently, I have a 224 character type alias; I was originally fine with this being on one line however, with black switching on their experimental string processing soon, this will get split into properly sized lines.

**To Reproduce**
Split a quoted TypeAlias over multiple lines in any way you see fit.

**Expected behavior**
My multi-line quoted TypeAlias to be accepted.

**Screenshots or Code**
```py
TypeType: TypeAlias = "Type | Literal['Invalid', 'Individual', 'Multiseat', 'GameServer', 'AnonGameServer', 'Pending', 'ContentServer', 'Clan', 'Chat', 'ConsoleUser', 'AnonUser', 'Max', 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]"
```
Gets formatted to
```py
TypeType: TypeAlias = (
    "Enum | Literal['Invalid', 'Individual', 'Multiseat', 'GameServer', 'AnonGameServer', 'Pending', 'ContentServer',"
    " 'Clan', 'Chat', 'ConsoleUser', 'AnonUser', 'Max', 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]"
)
```
And then you get `error: Type annotations cannot span multiple string literals`.

**VS Code extension or command-line**
pyright 1.1.147

**Comment by erictraut** [comment]
Compound string literals are not supported in this context. The type checker needs to run the parser and semantic analyzer on the contents of these quoted strings, and it needs to report errors detected in the contents. Each error must have a well-defined text range so it can be reported to the user in context. If the annotation expression is split across different string literals, it's not clear how errors could generally be reported.

For example, consider the following, which contains an error (because `2` is not a valid type argument).
```python
x1: TypeAlias = "List[2]"
x2: TypeAlias = "Li" "st" "[" "2]"
```

In the case of `x1`, it's clear which text range the error should be reported against. In the case of `x2`, it's not clear because the expression is split across different string literal tokens.

Out of curiosity, why are you enclosing the entire RHS in a string literal? There's no need to do so unless portions are forward references. That's not required in your example.

```python
TypeType: TypeAlias = (
    Type
    | Literal[
        "Invalid",
        "Individual",
        "Multiseat",
        "GameServer",
        "AnonGameServer",
        "Pending",
        "ContentServer",
        "Clan",
        "Chat",
        "ConsoleUser",
        "AnonUser",
        "Max",
        0,
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
    ]
)
```

Even if portions of the type annotation refer to forward declarations, you can enclose those parts in quotes. You don't need to enclose the entire expression in quotes.

So my recommendation is to avoid enclosing type annotations in string literals where possible. Where it's required for forward references, use individual string literals for those portions.

**Comment by Gobot1234** [comment]
I'm using 3.10 syntax on 3.7+ where that snippet isn't valid at run time, I was just surprised that this wasn't supported as this is a case that PyCharm's type checker supports fine.

**Comment by erictraut** [comment]
I see. Yeah, you would need to stick with the older syntax for unions in this case.

Like I mentioned above, supporting concatenated strings in type annotations adds significant complexity and creates edge cases for error reporting that are not clear how to resolve. For these reasons, I'm reluctant to add support for this.

For now, I'm going to close this feature request. We could revisit this in the future if we receive sufficient feedback (upvotes) from other pyright/pylance users. 

In the meantime, there are straightforward workarounds.

**Comment by jakebailey** [comment]
Given the string is always concatenated, I'm surprised there would be any confusion over the diagnostic range for anything (but, I can see where editor features may get very confused).

**Comment by Gobot1234** [comment]
Ok if concatenation is a concern what about only supporting triple quoted strings?
```py
TypeType: TypeAlias = """
	Type
    | Literal[
        "Invalid",
        "Individual",
        "Multiseat",
        "GameServer",
        "AnonGameServer",
        "Pending",
        "ContentServer",
        "Clan",
        "Chat",
        "ConsoleUser",
        "AnonUser",
        "Max",
        0,
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
    ]
"""
```

**Comment by erictraut** [comment]
Good thinking. I was about to respond with this suggestion myself. Yes, this would be straightforward to support, and it doesn't introduce any edge cases with error reporting.

**Comment by erictraut** [comment]
I've added support for triple-quoted type annotations. This will be included in the next release.

**Comment by Gobot1234** [comment]
Thank you :)

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.167, which I just published. It will also be included in the next release of pylance.
